sklearn/linear_model/tests/test_base.py

```diff
@@ -232,7 +255,9 @@ def test_linear_regression_pd_sparse_dataframe_warning():
     assert hasattr(df, "sparse")
 
     with pytest.warns(None) as record:
+        reg = LinearRegression()
```

Comment:
No need to put it back here. In fact, a more correct way would be to edit the one above like so:

```py
reg = LinearRegression()
with pytest.warns(UserWarning, match=msg):
	reg.fit(df.iloc[:, 0:2], df.iloc[:, 3])

```

This way, it is clear that the warning is issued in fit and not during the construction of the object.

That's a nit though.